java/code/src/com/redhat/rhn/common/security/acl/Access.java

```diff
@@ -51,12 +51,14 @@
 public class Access extends BaseHandler {
 
     protected static final Logger LOG = Logger.getLogger(Access.class);
+    private final ClusterManager clusterManager;
 
     /**
      * Constructor for Access object
+     * @param clusterManagerIn
      */
-    public Access() {
-        super();
+    public Access(ClusterManager clusterManagerIn) {
```

Comment:
> I expect that we will be moving all the *Manager classes used in this class to the constructor as well and will modify the needed static methods from them.
 Do we have a strategy here that we will follow or will it be just Adhoc like we did it here for ClusterManager?

Yes that the plan. The current goal was still on removing all global references to `SaltService/SystemQuery/SaltApi`.
So the strategy is use is to go top down and refactor everything needed to get rid of a particular global reference and try to mostly ignore anything else.

> On a side note: One thing that looks odd or not so good is passing null in the constructor of Access at so many places in Test* just to satisfy the signature.
I know i think i will fix that now. I just did that to safe time because its obvious that `ClusterManager` is rarely even used by `Access`.